src/note_encryption.rs

```diff
@@ -97,14 +97,17 @@ impl memuse::DynamicUsage for OrchardDomain {
 impl OrchardDomain {
     /// Constructs a domain that can be used to trial-decrypt this action's output note.
     pub fn for_action<T>(act: &Action<T>) -> Self {
-        OrchardDomain {
-            rho: *act.nullifier(),
-        }
+        Self::for_nullifier(*act.nullifier())
     }
 
     /// Constructs a domain from a nullifier.
+    ///
+    /// The provided nullifier must be the nullifier revealed in the action of the note being
+    /// encrypted or decrypted.
     pub fn for_nullifier(nullifier: Nullifier) -> Self {
```

Comment:
This API was added by @AloeareV to resolve #339, but I feel that this should _also_ be replaced by `for_compact_action` (to ensure the correct nullifier is used).

Reply:
Here, we can definitely use `for_compact_action`.